.evergreen/build_all.sh

```diff
@@ -14,25 +14,14 @@ _cxxflags=""
 
 : "${CONFIGURE_ONLY:=}"
 
-# Use C driver helper script to find cmake binary, stored in $CMAKE.
 if [ "$OS_NAME" = "windows" ]; then
-    : "${CMAKE:=/cygdrive/c/cmake/bin/cmake}"
     # Enable exception handling for MSVC
     _cxxflags="-EHsc"
-    if [ "${WINDOWS_32BIT-}" != "ON" ]; then
+    if test "${WINDOWS_32BIT-}" != "ON" && ! "${USE_NINJA-false}" ; then
```

Comment:
I also realized that executing an arbitrary command from an environment variable is just really suspicious altogether. But that led me to another idea:

```bash
is_true() {
    declare var="$1"
    declare val="${!var-@L}"  # Convert to lowercase, default is '' empty
    set -x
    case "$val" in
    1|true|yes|on)
        return 0;;
    0|false|no|off|"")
        return 1;;
    *)
        log "Unknown 'boolean' value for \$$var: '$val'"
        return 2;;
    esac
}

...

if is_true USE_NINJA; then 
  ...
fi
```

Which affords lenience in how the booleans are spelled, and no longer requires that each boolean be specially handled to prevent tripping `nounset`. Too clever?

Reply:
I like that approach much better.  The only thing I would say is missing is a return trap that executes `set +x`, since we don't want xtrace to stay on after this function call.